public/app/plugins/panel/table/module.tsx

```diff
@@ -47,10 +47,26 @@ export const plugin = new PanelPlugin<Options, CustomFieldConfig>(TablePanel)
               { value: TableCellDisplayMode.LcdGauge, label: 'LCD gauge' },
               { value: TableCellDisplayMode.BasicGauge, label: 'Basic gauge' },
               { value: TableCellDisplayMode.JSONView, label: 'JSON View' },
+              { value: TableCellDisplayMode.Popup, label: 'Popup' },
               { value: TableCellDisplayMode.Image, label: 'Image' },
             ],
           },
         })
+        .addSelect({
+          path: 'popupDisplayMode',
```

Comment:
To support cell display modes with options I think the best think would be to change the display mode into an object so that each display mode can have a custom editor and sub options. 

Created an issue for that: https://github.com/grafana/grafana/issues/31315